x-pack/legacy/plugins/ml/public/data_frame/common/job.ts

```diff
@@ -54,24 +56,46 @@ export const refreshTransformList$ = new BehaviorSubject<REFRESH_TRANSFORM_LIST_
   REFRESH_TRANSFORM_LIST_STATE.IDLE
 );
 
-export const useRefreshTransformList = () => {
-  const [isLoading, setIsLoading] = useState(false);
-  const [isRefresh, setIsRefresh] = useState(false);
-
+export const useRefreshTransformList = (
+  callback: {
+    isLoading?(d: boolean): void;
+    onRefresh?(): void;
+  } = {}
+) => {
   useEffect(() => {
-    const sub = refreshTransformList$.subscribe(s => {
-      setIsLoading(s === REFRESH_TRANSFORM_LIST_STATE.LOADING);
-      setIsRefresh(s === REFRESH_TRANSFORM_LIST_STATE.REFRESH);
-    });
-    return () => sub.unsubscribe();
+    const distinct$ = refreshTransformList$.pipe(distinctUntilChanged());
+
+    const subscriptions: Subscription[] = [];
+
+    if (typeof callback.onRefresh !== 'undefined') {
+      // initial call to refresh
+      callback.onRefresh();
+
+      subscriptions.push(
+        distinct$
+          .pipe(filter(state => state === REFRESH_TRANSFORM_LIST_STATE.REFRESH))
+          .subscribe(() => typeof callback.onRefresh !== 'undefined' && callback.onRefresh())
+      );
+    }
+
+    if (typeof callback.isLoading !== 'undefined') {
+      subscriptions.push(
+        distinct$.subscribe(
+          state =>
+            typeof callback.isLoading !== 'undefined' &&
```

Comment:
is this `typeof` check needed if it's already been checked a few lines above.

Reply:
Fixed in 96782e2.

Reply:
I tried without but TypeScript said no 😺